File: client/src/api/receipts.ts
import { api } from './axios';
import type { Sale } from '@/types/sale';

export interface ReceiptResponse {
  business: {
    name: string;
    logoUrl: string | null;
    address: string | null;
    phone: string | null;
  };
  sale: Sale;
  footer: string;
}

export const receiptApi = {
  get: (saleId: string) =>
    api
      .get<{ data: ReceiptResponse }>(`/client/receipts/${saleId}`)
      .then((r) => r.data.data),

  pdf: (saleId: string) =>
    api
      .get<{ data: { url: string | null; message?: string } }>(
        `/client/receipts/${saleId}/pdf`
      )
      .then((r) => r.data.data),

  email: (saleId: string, to: string, customerName?: string) =>
    api
      .post(`/client/receipts/${saleId}/email`, { to, customerName })
      .then((r) => r.data.data),
};

File: Android/client/lib/api/receipts.dart
import 'api_client.dart';

class ReceiptsApi {
  static final _api = BizOsApiClient.instance;

  static Future<Response<dynamic>> get(String saleId) =>
      _api.get('/client/receipts/$saleId');
  static Future<Response<dynamic>> pdf(String saleId) =>
      _api.get('/client/receipts/$saleId/pdf');
  static Future<Response<dynamic>> email(
    String saleId,
    Map<String, dynamic> data,
  ) => _api.post('/client/receipts/$saleId/email', data: data);
}
